streams_map.go

```diff
@@ -69,14 +71,42 @@ func (m *streamsMap) GetOrOpenStream(id protocol.StreamID) (*stream, error) {
 		return s, nil // s may be nil
 	}
 
-	// ... we don't have an existing stream, try opening a new one
+	// ... we don't have an existing stream
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 	// We need to check whether another invocation has already created a stream (between RUnlock() and Lock()).
 	s, ok = m.streams[id]
 	if ok {
 		return s, nil
 	}
+
+	if id <= m.highestStreamOpenedByPeer {
+		return nil, nil
+	}
+
+	highestOpened := m.highestStreamOpenedByPeer
+	sid := id
+	for sid > highestOpened {
```

Comment:
I wanted to rewrite this loop anyway, such that streams are opened in order. It just seemed easier to me to implement this way first.